scikeras/utils/transformers.py

```diff
@@ -0,0 +1,197 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List, Union
+
+import numpy as np
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import FunctionTransformer, OneHotEncoder, OrdinalEncoder
+from sklearn.utils.multiclass import type_of_target
+from tensorflow.python.keras.losses import is_categorical_crossentropy
+
+
+class Ensure2DTransformer(TransformerMixin, BaseEstimator):
+    """Transforms from 1D -> 2D and back.
+    """
+
+    def fit(self, X: np.ndarray) -> "Ensure2DTransformer":
+        self.should_transform_ = X.ndim == 1
+        return self
+
+    def transform(self, X: np.ndarray) -> np.ndarray:
+        if self.should_transform_:
+            if X.ndim != 1:
+                raise ValueError(
+                    "Expected a 1D array for `X`." f" Got a {X.ndim}D array instead."
+                )
+            X = X.reshape(-1, 1)
+        return X
+
+    def inverse_transform(self, X: np.ndarray) -> np.ndarray:
+        if self.should_transform_:
+            if X.ndim != 2:
+                raise ValueError(
+                    f"Expected a 2D array for `X`, but got a"
+                    f" {X.ndim}D array instead."
+                )
+            if X.shape[1] != 1:
+                raise ValueError(
+                    f"Expected `X.shape[1] == 1`"
+                    f" but got `X.shape[1] == {X.shape}` instead."
+                )
+            X = np.squeeze(X, axis=1)
+        return X
+
+
+class ClassifierLabelEncoder(BaseEstimator, TransformerMixin):
+    """Default target transformer for KerasClassifier.
+    """
+
+    def __init__(self, loss=None):
+        self.loss = loss
+
+    def fit(self, X: np.ndarray) -> "ClassifierLabelEncoder":
+        y = X  # rename for clarity, the input is always expected to be a target `y`
+        target_type = type_of_target(y)
+        encoders = {
+            "binary": make_pipeline(
+                Ensure2DTransformer(), OrdinalEncoder(dtype=np.float32),
+            ),
+            "multiclass": make_pipeline(
+                Ensure2DTransformer(), OrdinalEncoder(dtype=np.float32),
```

Comment:
Why is float32 hardcoded here? Does it rely on Keras defaulting to float32?

Reply:
I guess we should be checking `tf.keras.backend.floatx()`. The idea is that since we're doing a dtype conversion here, we might as well make it to whatever Keras is using.

Reply:
Changed the dtype to `tf.keras.backend.floatx()`.